core/domain/topic_services_test.py

```diff
@@ -186,6 +276,178 @@ def test_raises_error_while_computing_topic_summary_with_invalid_data(
+
+    def test_generate_topic_summary_when_publishing_story_chapter(
+        self
+    ) -> None:
+        self._create_linked_explorations(
+            self.TOPIC_ID, self.story_id_1, ['exp_1'])
+        topic_services.publish_story(
+            self.TOPIC_ID, self.story_id_1, self.user_id_admin)
+        story = story_fetchers.get_story_by_id(self.story_id_1)
+
+        with self.swap_with_checks(
+            topic_services, 'generate_topic_summary',
+            topic_services.generate_topic_summary,
+            expected_args=[(self.TOPIC_ID,)]
+        ):
+            topic_services.update_story_and_topic_summary(
+                self.user_id_admin, self.story_id_1,
+                [story_domain.StoryChange({
+                    'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY, 
+                    'property_name': story_domain.STORY_NODE_PROPERTY_STATUS,
+                    'node_id': story.story_contents.nodes[0].id,
+                    'old_value': constants.STORY_NODE_STATUS_DRAFT,
+                    'new_value': constants.STORY_NODE_STATUS_PUBLISHED
+                })], 'Publish story chapter.', self.TOPIC_ID)
+
+    def test_generate_topic_summary_when_unpublishing_story_chapter(
+        self
+    ) -> None:
+        self._create_linked_explorations(
+            self.TOPIC_ID, self.story_id_1, ['exp_1'])
+        topic_services.publish_story(
+            self.TOPIC_ID, self.story_id_1, self.user_id_admin)
+        story = story_fetchers.get_story_by_id(self.story_id_1)
+        topic_services.update_story_and_topic_summary(
+            self.user_id_admin, self.story_id_1,
+            [story_domain.StoryChange({
+                'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY, 
+                'property_name': story_domain.STORY_NODE_PROPERTY_STATUS,
+                'node_id': story.story_contents.nodes[0].id,
+                'old_value': constants.STORY_NODE_STATUS_DRAFT,
+                'new_value': constants.STORY_NODE_STATUS_PUBLISHED
+            })], 'Publish story chapter.', self.TOPIC_ID)
+
+        with self.swap_with_checks(
+            topic_services, 'generate_topic_summary',
+            topic_services.generate_topic_summary,
+            expected_args=[(self.TOPIC_ID,)]
+        ):
+            topic_services.update_story_and_topic_summary(
+                self.user_id_admin, self.story_id_1,
+                [story_domain.StoryChange({
+                    'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY, 
+                    'property_name': story_domain.STORY_NODE_PROPERTY_STATUS,
+                    'node_id': story.story_contents.nodes[0].id,
+                    'old_value': constants.STORY_NODE_STATUS_PUBLISHED,
+                    'new_value': constants.STORY_NODE_STATUS_DRAFT
+                })], 'Unpublish story chapter.', self.TOPIC_ID)
+
+    def test_generate_topic_summary_when_deleting_published_story_chapter(
```

Comment:
For these tests, in addition to checking whether the generate() method is called, do we also need to verify the correctness of the newly-generated topic summary (e.g. to make sure it doesn't include old stuff)? In general validating return values is a stronger test than validating whether something is called and gives better integration/end-to-end coverage of the whole system.

Reply:
Done.